packages/containerd/2001-selinux-add-DefaultLabels-helper.patch

```diff
@@ -0,0 +1,86 @@
+From 75ef4d007f7c4db0805a127d56d3347970f48a59 Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Wed, 29 Apr 2020 15:10:04 +0000
+Subject: [PATCH 2001/2002] selinux: add DefaultLabels helper
+
+This is very similar to the existing InitLabels helper function, but
+avoids allocating a new MCS label pair which would isolate containers
+in a pod from each other and break compatibility.
+
+In v1.4.0, containerd will have support for ensuring that containers
+in a pod end up sharing the MCS label pair, which should address the
+compatibility problem.
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ .../selinux/go-selinux/label/label.go         |  6 ++++
+ .../selinux/go-selinux/label/label_selinux.go | 36 +++++++++++++++++++
+ 2 files changed, 42 insertions(+)
+
+diff --git a/vendor/github.com/opencontainers/selinux/go-selinux/label/label.go b/vendor/github.com/opencontainers/selinux/go-selinux/label/label.go
+index e178568..194c797 100644
+--- a/vendor/github.com/opencontainers/selinux/go-selinux/label/label.go
++++ b/vendor/github.com/opencontainers/selinux/go-selinux/label/label.go
+@@ -2,6 +2,12 @@
+ 
+ package label
+ 
++// DefaultLabels returns the process and file labels that should be used on the
++// system if nothing more specific is requested.
++func DefaultLabels() (string, string, error) {
++	return "", "", nil
++}
++
+ // InitLabels returns the process label and file labels to be used within
+ // the container.  A list of options can be passed into this function to alter
+ // the labels.
+diff --git a/vendor/github.com/opencontainers/selinux/go-selinux/label/label_selinux.go b/vendor/github.com/opencontainers/selinux/go-selinux/label/label_selinux.go
+index 2730fcf..f2f0ff0 100644
+--- a/vendor/github.com/opencontainers/selinux/go-selinux/label/label_selinux.go
++++ b/vendor/github.com/opencontainers/selinux/go-selinux/label/label_selinux.go
+@@ -23,6 +23,42 @@ var validOptions = map[string]bool{
+ 
+ var ErrIncompatibleLabel = fmt.Errorf("Bad SELinux option z and Z can not be used together")
+ 
++// DefaultLabels returns the process and file labels that should be used on the
++// system if nothing more specific is requested.  The labels returned will not
++// include a random MCS string that is guaranteed to be unique, and do not need
++// any subsequent cleanup.
++func DefaultLabels() (plabel string, mlabel string, Err error) {
++	processLabel, mountLabel := selinux.ContainerLabels()
++	if processLabel == "" {
++		return "", "", nil
++	}
++
++	defer func() {
++		if Err != nil {
++			selinux.ReleaseLabel(processLabel)
++		}
++	}()
++
++	pcon, err := selinux.NewContext(processLabel)
++	if err != nil {
++		return "", "", err
++	}
++
++	mcon, err := selinux.NewContext(mountLabel)
```

Comment:
I would expect us to want to clean up `mcon` as well.  There are no early-exits after this point in the function, but this can become a trap for future changes.

(see comment at the top of this file for context)